redis_websocket_api/handler.py

```diff
@@ -150,9 +150,10 @@ async def listen(self):
 
     async def close(self):
         """Close all connections and cancel all tasks."""
-        asyncio.wait_for(  # attempt to say goodbye to the client
-            self.websocket.close(), self.read_timeout
-        )
+        if self.websocket.open:
+            await asyncio.wait_for(  # attempt to say goodbye to the client
```

Comment:
Was not changed in this MR but maybe it is relevant: In the [websockets documentation](https://websockets.readthedocs.io/en/stable/reference/server.html#websockets.server.WebSocketServerProtocol.close), it says that canceling a `close` is discouraged and one should change `close_timeout` if it takes too long. See [here](https://websockets.readthedocs.io/en/stable/reference/common.html#websockets.legacy.protocol.WebSocketCommonProtocol) for how `close_timeout` is interpreted.

(I hope I found the correct `close` method in the docs)

Reply:
Yes you are right, cancelling `close` skips quite a bit of cleanup code. I simplified the code on our end to let `websockets` deal with all that